frontend/src/testing/pushapi-mock.js

```diff
@@ -1,28 +1,21 @@
 /**
- * @author Stefan Dimitrov (stefan.dimitrov@clouway.com).
- */
+  * @author Stefan Dimitrov (stefan.dimitrov@clouway.com).
+  */
+ angular.module('clouway-push', [])
 
-angular.module('clouway-push')
+   .config(function ($provide, pushApiProvider, $httpBackend) {
```

Comment:
I think the following will be a simpler approach:
- If the backendServiceUrl is not set or empty string, then do not attempt to send any requests;
- we can have this config block in the `pushapi-mocks.js`:
  ```js
.config(function ($provide, pushApiProvider) {
     pushApiProvider.backendServiceUrl(''); // ensure no requests will be sent

     $provide.decorator('pushApi', function ($delegate) {
       $delegate.openConnection = angular.noop; // as before, no need to execute this logic in user's tests. If needed it can be spied on.
       return $delegate;
     });

   });
```